### PR TITLE
Return error message if supplied auth connector name doesn't match registered names.

### LINKED ISSUE #11427: SSO connector name is case-sensitive
### Description

SSO connector names are case-sensitive.

This is not a problem when using the Web UI (you just click on the SSO connector button). However, from the CLI the customer tried to run `tsh login --auth=okta` and it simply failed without indicating why. The reason was the connector name was `Okta` and the customer had to type `tsh login --auth=Okta`.

Customer had to read Teleport source to figure out the issue.

### PR DESCRIPTION
**Pull request by hatched**
If a supplied auth connector name does not match any of the registered names then return an error message with the possible valid connector names.

```
$ tsh --auth=Github login
ERROR: invalid connector name: Github; valid options: github, hatch
```

Fixes #11427

### CODE REVIEW

File: api/client/webclient/webclient.go [hunk 1]
```diff
@@ -185,8 +185,14 @@ func Ping(cfg *Config) (*PingResponse, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
 	defer resp.Body.Close()
+	if resp.StatusCode == 400 {
```
Comment (zmb3) [inline api/client/webclient/webclient.go:189]: There should be a constant in `package net/http` so you don't have to hard-code the 400.

File: api/client/webclient/webclient.go [hunk 2]
```diff
@@ -185,8 +185,14 @@ func Ping(cfg *Config) (*PingResponse, error) {
...
+		per := &PingErrorResponse{}
+		if err := json.NewDecoder(resp.Body).Decode(per); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return nil, fmt.Errorf(per.Error.Message)
```
Comment (zmb3) [inline api/client/webclient/webclient.go:194]:
```suggestion
		return nil, errors.New(per.Error.Message)
```

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -821,30 +821,57 @@ func (h *Handler) pingWithConnector(w http.ResponseWriter, r *http.Request, p ht
...
+	var collectorNames []string
+	for _, value := range oidcConnectors {
+		collectorNames = append(collectorNames, value.GetMetadata().Name)
+	}
+	for _, value := range samlConnectors {
+		collectorNames = append(collectorNames, value.GetMetadata().Name)
+	}
+	for _, value := range githubConnectors {
+		collectorNames = append(collectorNames, value.GetMetadata().Name)
 	}
```
Comment (zmb3) [inline lib/web/apiserver.go:869]: You could do the append in your existing loops above so that you don't have to loop through them a second time.

**Review by zmb3 — APPROVED** [review]

**Review by atburke — APPROVED** [review]

File: lib/web/apiserver.go [hunk 4]
```diff
@@ -820,10 +820,15 @@ func (h *Handler) pingWithConnector(w http.ResponseWriter, r *http.Request, p ht
 		return response, nil
```
Comment (tony-Rdz) [inline lib/web/apiserver.go:820]: api/client/webclient/webclient.go

**Pull request merged**